xls/dslx/stdlib/std.x

```diff
@@ -396,6 +396,41 @@ fn round_up_to_nearest_test() {
     assert_eq(u32:4, round_up_to_nearest(u32:4, u32:2));
 }
 
+// Returns `x` rounded up to `y`, where `y` is a positive power of 2
+fn round_up_to_pow2_unsigned<N: u32>(x: uN[N], y: uN[N]) -> uN[N] { (x + y - uN[N]:1) & -y }
```

Comment:
would something like https://graphics.stanford.edu/~seander/bithacks.html#RoundUpPowerOf2 be more efficient? (I think it should only result in bit slicing and +1 increment)

Reply:
Bit slices are possible only with a constant start/size. I cannot think of constant boundaries if we have `y` as an argument. Do you have any specific solution in mind? We can also move `y` to a generic parameter to make the slice start/size a compile constant. In that case, we can remove the need for one adder:
```rust
fn round_up_to_pow2_unsigned<Y: u32, N: u32, Y_CLOG2: u32 = { std::clog2(Y) }>(x: uN[N]) -> uN[N] {
    let base = x[Y_CLOG2 as s32:];
    let reminder = x[0:Y_CLOG2 as s32] != bits[Y_CLOG2]:0;
    (base as uN[N] + reminder as uN[N]) << Y_CLOG2
}
```

Reply:
I was thinking of something like this assuming you just need pow2 w/o the rounding:
```
fn round_up_to_pow2<N: u32, N_CLOG2_PLUS_1: u32 = { std::clog2(N) + u32:1 }>(v: uN[N]) -> uN[N] {
  let v = for (i, v) in u32:0..N_CLOG2_PLUS_1 {
    v | (v >> (uN[N]:1 << i))
  } (v-uN[N]:1);
  v+uN[N]:1
}
```

Reply:
I think the current bit twiddling form makes sense. When there's dynamic offsets as we've said you effectively need to either bit twiddle or call helper functions in the stdlib.